tests/validators/test_url.py

```diff
@@ -10,28 +12,80 @@
 
 def test_url_ok(py_and_json: PyAndJson):
     v = py_and_json(core_schema.url_schema())
-    url: Url = v.validate_test('https://example.com/foo/bar?baz=qux#quux')
-    assert isinstance(url, Url)
-    assert str(url) == 'https://example.com/foo/bar?baz=qux#quux'
-    assert repr(url) == "Url('https://example.com/foo/bar?baz=qux#quux')"
-    assert url.unicode_string() == 'https://example.com/foo/bar?baz=qux#quux'
-    assert url.scheme == 'https'
-    assert url.host == 'example.com'
-    assert url.unicode_host() == 'example.com'
-    assert url.path == '/foo/bar'
-    assert url.query == 'baz=qux'
-    assert url.query_params() == [('baz', 'qux')]
-    assert url.fragment == 'quux'
-    assert url.username is None
-    assert url.password is None
-    assert url.port == 443
+    urls: List[Url] = [
```

Comment:
better to parametrize the test than use a loop. Failures in loops in tests can be confusing to debug.

Reply:
but in this case, I don't think you need to test `Url` with the `py_and_json` parameterization, so probably better to just create a new test.

Reply:
Sounds good, created a new test with pretty much the same contents